app/scripts/controllers/transactions/index.js

```diff
@@ -1471,24 +1471,17 @@ export default class TransactionController extends EventEmitter {
     const signedEthTx = await this.signEthTx(
       unsignedEthTx,
       fromAddress,
-      ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
-      txMeta.custodyStatus ? txMeta : undefined,
-      ///: END:ONLY_INCLUDE_IN
+      this._shouldDisablePublish(txMeta) ? txMeta : undefined,
     );
 
-    ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
-    if (txMeta.custodyStatus) {
-      txMeta.custodyId = signedEthTx.custodian_transactionId;
-      txMeta.custodyStatus = signedEthTx.transactionStatus;
-
+    if (this._shouldDisablePublish(txMeta, signedEthTx)) {
```

Comment:
This is the true should publish check, but as it has logic also maybe a better name would be an optional `beforePublish`?

It should update the metadata, and invoke the `transactionUpdateController` inside the callback, and then we can use the boolean return value of that callback to know whether to `return` or not.